app/src/main/java/com/example/perfectweatherallyear/ui/weekWeather/WeekWeatherFragment.kt

```diff
@@ -5,20 +5,29 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
 import androidx.fragment.app.viewModels
 import androidx.recyclerview.widget.LinearLayoutManager
 import com.example.perfectweatherallyear.appComponent
+import com.example.perfectweatherallyear.changeFragment
 import com.example.perfectweatherallyear.databinding.FragmentWeekWeatherBinding
 import com.example.perfectweatherallyear.model.DayWeather
+import com.example.perfectweatherallyear.ui.detailWeather.ARG_DAY_WEATHER
+import com.example.perfectweatherallyear.ui.detailWeather.DetailWeatherFragment
 import com.example.perfectweatherallyear.ui.weekWeather.WeatherForecastAdapter.OnItemClick
 import com.example.perfectweatherallyear.util.NotificationUtil
+import com.google.gson.GsonBuilder
 import javax.inject.Inject
 
 class WeekWeatherFragment : Fragment() {
     private var weekWeatherList: List<DayWeather> = listOf()
     private lateinit var weatherForecastAdapter: WeatherForecastAdapter
     private lateinit var binding: FragmentWeekWeatherBinding
-    private val listener: OnItemClick = OnItemClickImpl()
+    private val listener: OnItemClick = object : OnItemClick {
+        override fun onItemClick(dayWeather: DayWeather, parentFragmentManager: FragmentManager) {
+            onAdaptorItemClick(dayWeather)
```

Comment:
Here you can see that there is no need in parentFragmentManager as a argument for onItemClick